### ISSUE TITLE
Lint errors when using new CLJS :npm-deps

### ISSUE THREAD

**Issue by rgdelato** [candid82/joker#11]
A recent version of ClojureScript introduced the :npm-deps compiler flag ([blog post](https://anmonteiro.com/2017/03/requiring-node-js-modules-from-clojurescript-namespaces/)). The example used in the blog post is:

```clojure
;; src/example/core.cljs
(ns example.core
  (:require left-pad))

(enable-console-print!)

;; pad the number 42 with five zeros
(println (left-pad 42 5 0))
```

In joker's lint mode, this currently throws two errors: One for the unused namespace and one at the site of the function call (unable to resolve symbol).

P.S. I'm still learning my way around ClojureScript and I've integrated joker's linter into my editor and it's helping quite a bit. Thank you!

**Comment by candid82** [comment]
Looks like the culprit is not `:npm-deps` per say but the fact that required JS module is being called directly, as if it was a function. I believe this functionality was introduced [here](https://github.com/clojure/clojurescript/commit/1d38f73a86081ad54cb230c507fbae183d768d6b).

This might be tricky to support since it's not legal in Clojure proper and is more like a special case in ClojureScript. I'll see what  I can do though.

**Comment by rgdelato** [comment]
Another kinda related issue is when using Javascript dependencies from [CLJSJS](https://github.com/cljsjs/packages), it's common practice to require an unused namespace that adds stuff to the global JS object, such as:

```clojure
(ns your.namespace
  (:require [cljsjs.react]))

(js/React.createElement "h1" "Hello there!")
```

**Comment by candid82** [comment]
This second issue is addressed in [v0.8.1](https://github.com/candid82/joker/releases/tag/v0.8.1) - see the note about `:ignored-unused-namespaces` config option.

**Comment by candid82** [comment]
Fix by https://github.com/candid82/joker/commit/f8d380c53f22c1f2b40f365a3801f4d45178b2c4
